presto-main/src/main/java/com/facebook/presto/sql/planner/plan/Assignments.java

```diff
@@ -164,6 +164,27 @@ public int size()
         return assignments.size();
     }
 
+    @Override
+    public boolean equals(Object o)
```

Comment:
Good point, though planer is using symbol references and each symbol is unique within the plan so it should be safe.